### ISSUE TITLE
Synced blocks: Define them as such in the Inspector

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#43261]
Related to https://github.com/WordPress/gutenberg/issues/42482 and https://github.com/WordPress/gutenberg/issues/32163.

When you select a template part or reusable block, it might be nice to clarify their global / synced nature – and perhaps outline usage – in the Inspector.

**Comment by jameskoster** [comment]
A rough idea:

<img width="413" alt="Screenshot 2022-08-16 at 11 02 53" src="https://user-images.githubusercontent.com/846565/184853300-b5e2cd67-3a79-46d4-ad44-5b9b8037490c.png">

Perhaps clicking the pill could open a modal that explains the concept in mode detail:

<img width="1005" alt="Screenshot 2022-08-16 at 11 04 01" src="https://user-images.githubusercontent.com/846565/184853524-52dd731f-75c1-46ff-9fc7-7792f7d83299.png">

**Comment by carolinan** [comment]
I am hesitant about introducing yet another term, "Synced blocks".  Can the features of these blocks be described without it?
I feel like "Used in 4 X" Where the X represents the actual item -posts, page, template etc, would be easier to understand?

We should consider that this is also true for the site title, tagline, logo/site icon.

**Comment by jameskoster** [comment]
I'm sure there's room to explore the copy! It might be tricky to use absolute terms though, since things like Reusable Blocks might appear in posts, pages, templates, template parts... the string might get a bit long 😅

> We should consider that this is also true for the site title, tagline, logo/site icon.

Good point. Navigation too.

**Comment by mtias** [comment]
Inferring how much a block is used throughout a site is not a trivial task, so I'd suggest a description that avoid saying that.

**Comment by jameskoster** [comment]
Since synced blocks are `post`s in their own right, could basic usage be saved as meta data to make it more trivial?

If not, it would be nice to do _something_ to differentiate between local/global blocks, as it's a pretty big pain point right now. It could be something like this, or https://github.com/WordPress/gutenberg/issues/32163, or a combination :)

**Comment by mtias** [comment]
Not really, the problem is figuring out how to keep it updated. Every time you embed a reusable block you'd bump the usage? What happens when it gets removed? What if you duplicate a post, or unpublish a page, or remove a template that referenced one? It gets unwieldy quickly. The right way to do it would be to query for its presence with `has_blocks`, which would need to be done performantly and on some cron job. Not a trivial thing to design well.

I don't think it adds that much context, anyways. A simpler "This block is synced" seems it fulfills mostly the same purpose.

**Comment by jameskoster** [comment]
> Every time you embed a reusable block you'd bump the usage?

I was thinking the 'count' would be updated when the parent document is updated / published. But yes I suppose that will be tricky considering imports etc.

> A simpler "This block is synced" seems it fulfills mostly the same purpose.

I agree, this would be a worthwhile update on its own.

**Comment by getdave** [comment]
So I took a look at this in the context of the Nav block. I realised that a few places in the codebase define `isSynced` as being `isReusableBlock` or `isTemplatePart`. I'm wondering if a better proxy might be `areInnerBlocksControlled` from `@wordpress/block-editor`?

That way it could apply generically to any block which is sync'd with an entity.

**Comment by getdave** [comment]
I spun up [a quick experimental PR to see how this might work if it applied to all blocks which control their inner blocks](https://github.com/WordPress/gutenberg/pull/46412)

**Comment by jameskoster** [comment]
Thanks for taking a look. 

This is where the fundamentally different behaviour of the Navigation and Template Part blocks makes things a bit awkward. 

With template parts (and reusable blocks), their appearance in the UI is informed by the referenced content. IE when I insert my "Centered Header" template part, that's how I see it presented in the toolbar, list view, etc:

<img width="937" alt="Screenshot 2022-12-09 at 12 48 43" src="https://user-images.githubusercontent.com/846565/206706048-10147934-06cd-40be-be28-36384cb08fe4.png">

The template part 'wrapper' is obfuscated away, so in this context it makes sense to mark the blocks as synced.

As you know Navigation doesn't work like this. It's always presented as "Navigation", regardless of which menu is referenced. This makes me question whether we should mark the Navigation as synced, because it's not really true. It's the referenced menu that is synced, not the parent block. I worry it might be confusing to have multiple Navigation instances in a document that are all marked as synced despite having (potentially) different appearances.

If Navigation behaved like Template Parts this wouldn't be an issue, but I appreciate that will be a big can of worms :)

**Comment by scruffian** [comment]
> If Navigation behaved like Template Parts this wouldn't be an issue, but I appreciate that will be a big can of worms :)

What changes would you expect to see in the navigation block for it behave like a Template Part? Here are some I thought of:

1. Display the name of the selected navigation menu (if there is one) in the block toolbar
2. Do the same thing in the list view

What else am I missing?

**Comment by jameskoster** [comment]
I would consider the block name in the Inspector too, but that might be trickier for Navigation. Worth noting this doesn't currently happen for template parts which is a bug to fix (https://github.com/WordPress/gutenberg/issues/42154).

If there's a plan to support navigation editing in focus mode, then might be worth keeping an eye on too.

Edit: I forgot one, template parts and reusable blocks appear in the Inserter by name:

<img width="343" alt="Screenshot 2022-12-09 at 17 21 26" src="https://user-images.githubusercontent.com/846565/206757800-353b27a5-3b39-4db6-8209-0a68982808b1.png">

**Comment by talldan** [comment]
I've made a PR that shows the menu name in some of those areas - https://github.com/WordPress/gutenberg/pull/46457.

It also fixes https://github.com/WordPress/gutenberg/issues/42154 (I can extract that out to a separate patch if needed).

> I forgot one, template parts and reusable blocks appear in the Inserter by name:

That should be possible too, though some users may have a lot of menus. Happy to try it out in a another PR, though I don't think it'll be quite as easy as it was for template parts.

**Comment by getdave** [comment]
Thanks for the feedback folks. I'm glad the PR has sparked discussion.

Perhaps I should ask a different question? **What can we do design wise to make it more obvious that the contents of a given Navigation block are synced to an entity?**

The requirement for improvement is there and now we need to devise a solution.

I don't think we necessarily have to make it identical to how TPs or RBs appear. But some of the styling could be similar.

@scruffian has made some suggestions.

@SaxonF @jasmussen may also have a view?

**Comment by jasmussen** [comment]
I think #46438 helps, #46457 can as well though to a lesser extent. As Jay suggests, should we also apply the purple "synced" color? (Both for any clickthrough overlays applied, but also to the icon and select state in the inspector). Those would be some initial steps, and I imagine more pieces of the puzzle could fall into place as the inspector editing interface matures.

**Comment by getdave** [comment]
> As Jay suggests, should we also apply the purple "synced" color? (Both for any clickthrough overlays applied, but also to the icon and select state in the inspector).

That's doable. I can look into that. Still open to any additional thoughts as well.

Thank you to everyone who contributed here. It's much appreciated.

**Comment by jordesign** [comment]
Wondering if the way we present Synced Patterns in 6.3 and onwards is sufficient to close this issue @jameskoster  ?

<img width="1394" alt="Screenshot 2023-09-04 at 11 48 01 am" src="https://github.com/WordPress/gutenberg/assets/1842363/f1077019-cc83-4cd7-a6a5-9e45ab000bf9">

**Comment by jameskoster** [comment]
I don't think so because template parts are not captured. 

At some point we'll probably want to allow custom descriptions for synced patterns, so the pill still seems like the best approach to me. Perhaps with "Syncing: Fully synced" to echo the details panel:

<img width="281" alt="Screenshot 2023-09-04 at 11 19 13" src="https://github.com/WordPress/gutenberg/assets/846565/e528f805-7e0e-4566-b675-8408ff33467e">
